### PR TITLE
834 create custom action workflow to publish report

### PR DESCRIPTION
**Pull request by Ritik232**
## Changes Applied: 
Created a script which sends data to allure tool and generated a report. Also created a action.yml workflow to create a custom action for the script.

### CODE REVIEW

**Review by Nkashyap-calance — DISMISSED** [review]
LGTM

**Review by ypmittal — CHANGES_REQUESTED** [review]
code needs to pass js linting before i can review properly

From a cursory review, below improvements are needed:
- cookie parsing logic needs to be improved, preferably use some package
- error handling is incomplete. see example below. Missing return/exception to terminate the flow when API call fails.
```
                            if (report.status !== 200) {
                                core.setFailed(`Something went wrong in generating report. Status Code: ${report.status}`)
                            }
```
- nesting level in the code is too high
- make the linter ignore the dist file
- do not commit package-lock.json
- add details to README
- remove files `Dockerfile` & `build.yaml` which are not required for this repository

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+    "env": {
+        "browser": true,
```
Comment (ypmittal) [inline .eslintrc.json]: should not be true

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+description: Generate selenium report for the allure tool and send latest data.
+
+inputs:
+  project_name:
+    description: "This refers to the name of the project that goes in allure testing tool."
+    required: false
+    type: string
+    default: "${{ github.event.repository.name }}"
+  results_path:
+    description: "Path of the test results directory, relative to the test_suite_path. Defaults to test-results."
```
Comment (ypmittal) [inline action.yml]: change the description

File: action.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    required: false
+    type: string
+    default: "test-results"
+  username:
+    description: "Username for QA tool of admin"
```
Comment (ypmittal) [inline action.yml]: Username for the QA tool. Should have admin access.

File: action.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    required: true
+  password:
+    description: "Password for QA tool of admin"
```
Comment (ypmittal) [inline action.yml]: see above comment

File: index.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
+const core = require('@actions/core')
+const fs = require('fs/promises')
+const path = require('path')
+const fetch = require('node-fetch')
+
+const apiUrl = 'https://qa-reports.calance.work/allure-api/allure-docker-service'
```
Comment (ypmittal) [inline index.js]: take this as an input

File: index.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+const loginCredentials = {
+  username: core.getInput('username'),
+  password: core.getInput('password')
+}
+const project = core.getInput('project_name')
+
+async function checkApiStatus () {
```
Comment (ypmittal) [inline index.js]: rename to `publishReport`

File: index.js [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+      throw new Error(`Failed to generate report. ${JSON.stringify(generatedReprot, null, 2)}`)
+    } else {
+      console.log(generatedReprot)
+    }
+  } catch (error) {
+    core.setFailed(`Error occurred while hitting the API: ${error.message}`)
+  }
+}
+
+checkApiStatus().then()
```
Comment (ypmittal) [inline index.js]: you can remove then if nothing is being done inside it

**Review by ypmittal — APPROVED** [review]

**Pull request merged**